elastic/beats
# Add the cluster_uuid field to Enterprise search stats metricset

## What does this PR do?

This PR adds the `cluster_uuid` field to Enterprise search stats metricset to make it easier to correlate with health metricset events.

## Why is it important?

 This is a follow-up for the [recent PR](https://github.com/elastic/beats/pull/27549) that added the new module and we have just realized that having a uuid in all of our fields would make correlating them in Kibana a lot easier and make those events easier to work with in Kibana monitoring plugin (that relies on cluster UUID to group stack components together).

## Checklist


## Author's Checklist

- [ ] Tested it against the recent Enterprise Search snapshot that has the cluster_uuid field in Stats API responses.

## How to test this PR locally

* Deploy Enterprise Search
* Run `metricbeat` with the `enterprisesearch` module enabled
* Check `stats` events produced by the `enterprisesearch` module and you'll see the new field.

Alternatively, one can use integration tests to make sure the changes work (I've added data tests changes needed).

## Related issues

- Relates to https://github.com/elastic/beats/pull/27549
